src/indexer.h

```diff
@@ -11,7 +11,13 @@
 
 typedef struct {
     RedisModuleString *key;
-    RedisModuleString *value;
+    union {
```

Comment:
Im not a big fan of this, whats the benefits of using a union over just adding `valueList` and `listCount`?

Reply:
I think this complexity should reside in QueryPredicate and Label should be used only for the key's label storage.

Reply:
So Label will contain a kay and value, and the QueryPredicate will contain the valueList and listCount?